CoreAlgorithmPlugins/src/au/gov/asd/tac/constellation/plugins/algorithms/sna/centrality/PagerankCentralityPlugin.java

```diff
@@ -95,118 +92,181 @@ public PluginParameters createParameters() {
+            
+            if (PagerankVertex.delta < epsilon) {
+                break;
             }
 
-            final int linkCount = graph.getVertexLinkCount(vertexId);
-            for (int linkPosition = 0; linkPosition < linkCount; linkPosition++) {
-                final int linkId = graph.getVertexLink(vertexId, linkPosition);
-                final int linkLowId = graph.getLinkLowVertex(linkId);
-                final int linkHighId = graph.getLinkHighVertex(linkId);
-                if (linkLowId != linkHighId) {
-                    final int neighbourId = vertexId == linkLowId ? linkHighId : linkLowId;
-                    final int edgeCount = graph.getLinkEdgeCount(linkId);
-                    for (int edgePosition = 0; edgePosition < edgeCount; edgePosition++) {
-                        final int edgeId = graph.getLinkEdge(linkId, edgePosition);
-                        final int edgeDirection = graph.getEdgeDirection(edgeId);
-                        if ((treatUndirectedBidirectional && edgeDirection == GraphConstants.FLAT)
-                                || (vertexId < neighbourId && edgeDirection == GraphConstants.DOWNHILL)
-                                || (vertexId > neighbourId && edgeDirection == GraphConstants.UPHILL)) {
-                            int neighbourOutCount = 0;
-                            final int transactionCount = graph.getVertexTransactionCount(neighbourId);
-                            for (int transactionPosition = 0; transactionPosition < transactionCount; transactionPosition++) {
-                                final int transactionId = graph.getVertexTransaction(neighbourId, transactionPosition);
-                                final int transactionDirection = graph.getTransactionDirection(transactionId);
-                                final int transactionSourceId = graph.getTransactionSourceVertex(transactionId);
-                                final int transactionDestinationId = graph.getTransactionDestinationVertex(transactionId);
-                                if (transactionSourceId != transactionDestinationId) {
-                                    final int otherId = neighbourId == transactionSourceId ? transactionDestinationId : transactionSourceId;
-                                    if ((treatUndirectedBidirectional && transactionDirection == GraphConstants.FLAT)
-                                            || (neighbourId < otherId && transactionDirection == GraphConstants.UPHILL)
-                                            || (neighbourId > otherId && transactionDirection == GraphConstants.DOWNHILL)) {
-                                        neighbourOutCount++;
-                                    }
-                                }
-                            }
+        }
+
+        // update the graph with pagerank values
+        PagerankVertex.commitPageranksToGraph();
+    }
+    
+    static class PagerankVertex {
```

Comment:
Could this be made private or was the intention that this could be called elsewhere?

Reply:
Ahh yes it definitely can :)